sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -477,11 +477,18 @@ abstract class GpuToTimestamp
     } else { // Timestamp or DateType
       lhs.getBase.asTimestampMicroseconds()
     }
-    withResource(tmp) { r =>
-      // The type we are returning is a long not an actual timestamp
-      withResource(Scalar.fromInt(downScaleFactor)) { downScaleFactor =>
-        withResource(tmp.asLongs()) { longMicroSecs =>
-          longMicroSecs.div(downScaleFactor)
+    // Return Timestamp value if dataType it is expecting is of TimestampType
+    if (dataType.equals(TimestampType)) {
+      tmp
+    }
+    else {
+      withResource(tmp) { r =>
+        // The type we are returning is a long not an actual timestamp
+        withResource(Scalar.fromInt(downScaleFactor)) { downScaleFactor =>
+          withResource(tmp.asLongs()) { longMicroSecs => {
```

Comment:
nit: This usage of `withResource` has an additional set of `{` and `}` after `=>` which is inconsistent with the other lines and could be omitted